jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/dao/RDSDBManager.java

```diff
@@ -207,4 +211,54 @@ public static void insertRecord(String tableName, Map<String, String> data) {
             LOGGER.error("Error Executing Query", e);
         }
     }
+
+    /**
+     * Checks if a plugin exists in the database.
+     *
+     * @param dataSource
+     * @return
+     */
+
+    public static boolean pluginExists(String dataSource) {
+        String recordsCountQuery = "SELECT COUNT(*) AS count FROM cf_PolicyTable WHERE assetGroup = ?";
+        try (Connection conn = getConnection();
+             PreparedStatement preparedStatement = conn.prepareStatement(recordsCountQuery)) {
+            preparedStatement.setString(1, dataSource);
+            return executeCountQuery(preparedStatement) > 0;
+        } catch (SQLException sqlException) {
+            LOGGER.error("Error executing query", sqlException);
+        }
+        return false;
+    }
+
+    /**
+     * Determines the status of the policy based on the existing status and the plugin existence.
+     *
+     * @param policy
+     * @param isPluginAlreadyExists
+     * @return
+     */
+    private static String determinePolicyStatus(PolicyTable policy, boolean isPluginAlreadyExists) {
+        if (!StringUtils.isNullOrEmpty(policy.getStatus())) { //Case 1 : When Policy has Status (Contrast) use that
```

Comment:
lets not add plugin names in the comments

Reply:
addressed them in latest push